Helper/Order.php

```diff
@@ -1072,10 +1082,15 @@ public function processExistingOrder($quote, $transaction)
         if ($order = $this->getExistingOrder(null, $quote->getId())) {
 
             if ($order->isCanceled()) {
+                if ($this->featureSwitches->isCancelFailedPaymentOrderInstadOfDeleting()
+                    && $order->getData('bolt_failed_payment')) {
```

Comment:
What if we just disabled the feature flag? Maybe the second condition is enough?

Reply:
And the second question here. Looks like we can't find the order with 'bolt_failed_payment', because in the same moment when we set this field, we reset quote_id.
Could you check?

Reply:
@vitaliyreznikov Great catch. Actually the bolt_failed_payment flag is redundant. Will post an update soon.

Reply:
done